pengine/allocate.c

```diff
@@ -2145,6 +2146,11 @@ stage8(pe_working_set_t * data_set)
 
     crm_log_xml_trace(data_set->graph, "created resource-driven action list");
 
+    /* adding pseudo action that carries list of nodes in maintenance */
+    crm_trace("adding pseudo action holding nodes in maintenance");
+    action = get_pseudo_op(CRM_OP_MAINTENANCE_NODES, data_set);
+    set_bit(action->flags, pe_action_print_always);
+
```

Comment:
was thinking of that as well but like this I can use the empty action to purge the maintenance-flag from all remote-lrmds.
Of course having a transition without maintenance-pseudo-action could purge the flags as well.
But this would either lead to the sequence becoming relevant and purging the flags multiple times when there are
other pseudo actions sent to remote_ra_process_pseudo.
Or the intelligence would have to be moved to te_actions.c.
Both has disadvantages (first performance and the sequence, 2nd intelligence at 2 points in the code).
And I had the feeling - without thinking it through in detail - it might be beneficial in a rolling upgrade scenario.